security/pkg/listwatch/listwatch.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Provides multiple namespace listerWatcher. This implementation is Largely from
+// https://github.com/coreos/prometheus-operator/pkg/listwatch/listwatch.go
+
+package listwatch
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+)
+
+// MultiNamespaceListerWatcher takes a list of namespaces and a
+// cache.ListerWatcher generator func and returns a single cache.ListerWatcher
+// capable of operating on multiple namespaces.
+func MultiNamespaceListerWatcher(namespaces []string, f func(string) cache.ListerWatcher) cache.ListerWatcher {
```

Comment:
Can you move this file to `pkg` in root directory so that all components can share it.

Reply:
Yes. I think it is reasonable. in order to reduce the API server pressure, it is better to allow each component to list/watch with specified namespaces. Any comments from @myidpt @incfly ?

Reply:
Might be. I'm okay with either way. But I think Pilot/Galley already reference some client libraries to listen to apiserver.

Reply:
I dont have strong opinion. My suggestion is to look at existing client libraries used in pilot/galley for service discovery before we try to hoist up to `pkg/` level.

Reply:
I think that makes sense. Suggest to put it in a separate PR so to reduce the reviewers involved in this PR.

Reply:
have a separated PR #11880